File: Code Generation/responses/python/deepseek/easy/maximum-students-on-a-single-bench.py
from typing import List

class Solution:
    def maxStudentsOnBench(self, students: List[List[int]]) -> int:
        if not students:
            return 0
        
        bench_students = {}
        
        for student_id, bench_id in students:
            if bench_id not in bench_students:
                bench_students[bench_id] = set()
            bench_students[bench_id].add(student_id)
        
        max_unique = 0
        for students_set in bench_students.values():
            max_unique = max(max_unique, len(students_set))
        
        return max_unique

File: Code Generation/responses/cpp/deepseek/easy/maximum-students-on-a-single-bench.cpp
#include <vector>
#include <unordered_map>
#include <unordered_set>

using namespace std;

class Solution {
public:
    int maxStudentsOnBench(vector<vector<int>> students) {
        unordered_map<int, unordered_set<int>> bench_to_students;
        
        for (const auto& record : students) {
            int student_id = record[0];
            int bench_id = record[1];
            bench_to_students[bench_id].insert(student_id);
        }
        
        int max_unique = 0;
        for (const auto& entry : bench_to_students) {
            int unique_count = entry.second.size();
            if (unique_count > max_unique) {
                max_unique = unique_count;
            }
        }
        
        return max_unique;
    }
};